scipy/optimize/_trustregion_constr/report.py

```diff
@@ -23,7 +23,7 @@ def print_iteration(cls, *args):
         # trust-constr typically provides a length 1 array. We have to coerce
         # it to a float, otherwise the string format doesn't work.
         args = list(args)
-        args[3] = float(args[3])
+        args[3] = args[3][0]
```

Comment:
Is this "typically" a 1d length-1 array, or is it _always_ 1d?

Reply:
I dug it out and found that `args[3]` is eventually the user-provided value, so it could be anything of size 1. I'll apply `item()`.